From 06e70817b828f23d4261988702ea4506871bddd5 Mon Sep 17 00:00:00 2001
From: bagyusz <phelyx92@gmail.com>
Date: Tue, 7 Oct 2014 15:33:35 +0200
Subject: [PATCH] 
 Settings-App2SD-Enable-moving-applications-to-an-external-volume

Change-Id: I468e0bac72a8d780cc7c038f0c96f9524231de1b
---
 src/com/android/settings/applications/InstalledAppDetails.java | 6 +++---
 src/com/android/settings/deviceinfo/StorageMeasurement.java    | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/com/android/settings/applications/InstalledAppDetails.java b/src/com/android/settings/applications/InstalledAppDetails.java
index 4755d3b..6f45f47 100644
--- a/src/com/android/settings/applications/InstalledAppDetails.java
+++ b/src/com/android/settings/applications/InstalledAppDetails.java
@@ -305,7 +305,7 @@ public class InstalledAppDetails extends Fragment
     }
 
     private void initMoveButton() {
-        if (Environment.isExternalStorageEmulated()) {
+        if (!Environment.isExternalAppsAvailableAndMounted()) {
             mMoveAppButton.setVisibility(View.INVISIBLE);
             return;
         }
@@ -492,7 +492,7 @@ public class InstalledAppDetails extends Fragment
         mExternalCodeSize = (TextView)view.findViewById(R.id.external_code_size_text);
         mExternalDataSize = (TextView)view.findViewById(R.id.external_data_size_text);
 
-        if (Environment.isExternalStorageEmulated()) {
+        if (!Environment.isExternalAppsAvailableAndMounted()) {
             ((View)mExternalCodeSize.getParent()).setVisibility(View.GONE);
             ((View)mExternalDataSize.getParent()).setVisibility(View.GONE);
         }
@@ -1045,7 +1045,7 @@ public class InstalledAppDetails extends Fragment
             mHaveSizes = true;
             long codeSize = mAppEntry.codeSize;
             long dataSize = mAppEntry.dataSize;
-            if (Environment.isExternalStorageEmulated()) {
+            if (!Environment.isExternalAppsAvailableAndMounted()) {
                 codeSize += mAppEntry.externalCodeSize;
                 dataSize +=  mAppEntry.externalDataSize;
             } else {
diff --git a/src/com/android/settings/deviceinfo/StorageMeasurement.java b/src/com/android/settings/deviceinfo/StorageMeasurement.java
index 34ef62b..e1dd3bb 100644
--- a/src/com/android/settings/deviceinfo/StorageMeasurement.java
+++ b/src/com/android/settings/deviceinfo/StorageMeasurement.java
@@ -249,7 +249,7 @@ public class StorageMeasurement {
                 long codeSize = stats.codeSize;
                 long dataSize = stats.dataSize;
                 long cacheSize = stats.cacheSize;
-                if (Environment.isExternalStorageEmulated()) {
+                if (!Environment.isExternalAppsAvailableAndMounted()) {
                     // Include emulated storage when measuring internal. OBB is
                     // shared on emulated storage, so treat as code.
                     codeSize += stats.externalCodeSize + stats.externalObbSize;
-- 
1.9.2

